Fix sidebar search, post bootstrap

kinow opened the pull request:
## Reasons for creating this PR

Bootstrap upgrade PR broke the sidebar search (completely forgot to test this part during the Bootstrap upgrade! Sorry :fearful: )

## Link to relevant issue(s), if any

- Closes #1311

## Description of the changes in this PR

Fix sidebar search, by matching previous CSS styles. Had to modify the structure of HTML, and copy the `btn-default` style from Bootstrap 3.

## Known problems or uncertainties in this PR

## Checklist

- [x] phpUnit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if not, explain why below)
- [x] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)

kinow commented:
This will need some testing locally or on finto dev. Here's what it looks like for me, running with Docker.

![image](https://user-images.githubusercontent.com/304786/165748821-0b2222a0-a522-43b7-b1cf-7c1be5e40334.png)

osma commented:
Thanks a lot @kinow ! Will test...

osma commented:
I tested this briefly. Here you can see the old Skosmos 2.14 (left) and new (right):

![kuva](https://user-images.githubusercontent.com/1132830/165758279-a5d3d3bc-af64-425f-8088-89ddbdd0bb0f.png)

It's working well now!

The font color is different for the options - it used to be blue (link color) but now it's black. I think the black color is better actually. These are not links so there is no reason to use blue color. Also the highlighting is different, but here too I think the new look is better.

For the group selection, I suggest changing the square checkboxes to round radio buttons, as they used to be. This is because only one selection is allowed. For the others, the square checkboxes are good.

In general there is a little bit more vertical spacing in the new layout, for example between the labels and the selection boxes (highlighted on picture). It's the same with the search results seen on the right - they take up more space than they used to. For example the headline "catacombs" is visible in the old layout on the left side but not in the new layout on the right side. I wonder if there is some general CSS setting (line-height maybe?) that is different in Bootstrap 5 than it used to be in Bootstrap 3? This isn't super critical, just something I wonder about because it seems to affect all page types, not just the search result page shown here.

kinow commented:
>For the group selection, I suggest changing the square checkboxes to round radio buttons, as they used to be. This is because only one selection is allowed. For the others, the square checkboxes are good.

:+1: 

>In general there is a little bit more vertical spacing in the new layout, for example between the labels and the selection boxes (highlighted on picture). It's the same with the search results seen on the right - they take up more space than they used to. For example the headline "catacombs" is visible in the old layout on the left side but not in the new layout on the right side. I wonder if there is some general CSS setting (line-height maybe?) that is different in Bootstrap 5 than it used to be in Bootstrap 3? This isn't super critical, just something I wonder about because it seems to affect all page types, not just the search result page shown here.

I will spend some time touching up after fixing the checkbox/radiobutton, trying to make it look more similar to finto. Can probably finish it tomorrow morning (not working, and my soccer team is playing in Sao Paulo, so eyes on the screen, ears to the radio :soccer: )

Thanks for testing it so quickly @osma!

kinow commented:
>For the group selection, I suggest changing the square checkboxes to round radio buttons, as they used to be. This is because only one selection is allowed. For the others, the square checkboxes are good.

Huh, interesting. Looking at the HTML DOM, the "By group" fields are actually using `type="radio"`.

![image](https://user-images.githubusercontent.com/304786/165832724-bdafb3af-a05f-4352-b084-0b700cde2651.png)

One CSS rule that I added to have sharp corners in the checkboxes ended up causing the radio inputs to have sharp corners too. Essentially becoming check boxes :laughing: Fixed. Now looking at the remaining style issues.

kinow commented:
>In general there is a little bit more vertical spacing in the new layout, for example between the labels and the selection boxes (highlighted on picture). It's the same with the search results seen on the right - they take up more space than they used to. For example the headline "catacombs" is visible in the old layout on the left side but not in the new layout on the right side. I wonder if there is some general CSS setting (line-height maybe?) that is different in Bootstrap 5 than it used to be in Bootstrap 3? This isn't super critical, just something I wonder about because it seems to affect all page types, not just the search result page shown here.

Bootstrap 5 had a class `.form-label` with `margin-bottom: .5em`. But Bootstrap 3 had a fixed margin of `5px`. I've copied the Bootstrap 3 setting over. The downside is when zooming in the margin won't be adjusted, which goes against accessibility practices, like WCAG. But now the look and feel should be more similar.

Some screenshots.

![image](https://user-images.githubusercontent.com/304786/165837202-c8756a40-295e-4369-9331-87abdb138395.png)

![image](https://user-images.githubusercontent.com/304786/165837244-bcf8c574-67fb-4758-8edd-7e50e032ae4e.png)

Had a look at the Feedback form and at the search language switcher to see if these changes broke those `<select>` inputs, but everything looks the same to me.

Ready for review again :tada: 
Thanks
Bruno

osma commented:
Excellent work @kinow! All good now, I'm merging this.
